prism-agent/service/server/src/main/scala/io/iohk/atala/pollux/credentialschema/SchemaRegistryServerEndpoints.scala

```diff
@@ -1,46 +1,47 @@
 package io.iohk.atala.pollux.credentialschema
 
+import io.iohk.atala.agent.walletapi.model.BaseEntity
 import io.iohk.atala.api.http.model.{Order, PaginationInput}
 import io.iohk.atala.api.http.{ErrorResponse, RequestContext}
 import io.iohk.atala.iam.authentication.Authenticator
-import io.iohk.atala.iam.authentication.apikey.ApiKeyEndpointSecurityLogic
+import io.iohk.atala.iam.authentication.Authorizer
+import io.iohk.atala.iam.authentication.DefaultAuthenticator
+import io.iohk.atala.iam.authentication.SecurityLogic
 import io.iohk.atala.pollux.credentialschema.SchemaRegistryEndpoints.*
 import io.iohk.atala.pollux.credentialschema.controller.CredentialSchemaController
 import io.iohk.atala.pollux.credentialschema.http.{CredentialSchemaInput, FilterInput}
 import io.iohk.atala.shared.models.WalletAccessContext
 import sttp.tapir.ztapir.*
 import zio.*
-import io.iohk.atala.agent.walletapi.model.Entity
 
 import java.util.UUID
 
 class SchemaRegistryServerEndpoints(
     credentialSchemaController: CredentialSchemaController,
-    authenticator: Authenticator
+    authenticator: Authenticator[BaseEntity] & Authorizer[BaseEntity]
 ) {
   def throwableToInternalServerError(throwable: Throwable) =
     ZIO.fail[ErrorResponse](ErrorResponse.internalServerError(detail = Option(throwable.getMessage)))
 
   val createSchemaServerEndpoint: ZServerEndpoint[Any, Any] =
     createSchemaEndpoint
-      .zServerSecurityLogic(ApiKeyEndpointSecurityLogic.securityLogic(_)(authenticator))
+      .zServerSecurityLogic(SecurityLogic.authorizeWith(_)(authenticator))
       .serverLogic {
-        case entity: Entity => { case (ctx: RequestContext, schemaInput: CredentialSchemaInput) =>
+        case wac => { case (ctx: RequestContext, schemaInput: CredentialSchemaInput) =>
           credentialSchemaController
             .createSchema(schemaInput)(ctx)
-            .provideSomeLayer(ZLayer.succeed(entity.walletAccessContext))
+            .provideSomeLayer(ZLayer.succeed(wac))
         }
       }
 
   val updateSchemaServerEndpoint: ZServerEndpoint[Any, Any] =
     updateSchemaEndpoint
-      .zServerSecurityLogic(ApiKeyEndpointSecurityLogic.securityLogic(_)(authenticator))
+      .zServerSecurityLogic(SecurityLogic.authorizeWith(_)(authenticator))
```

Comment:
Let's think in the next PR about the naming.
It looks like we do authorization with authenticator :)
authenticator should authenticate
authorizer should authorize

Reply:
For now the implementation is shared for authentication and authorization just to reduce the refactoring work., so yes, that variable is doing both :) as a consequence. Let's think about this after this PR.